internal/controlplane/handlers_repositories.go

```diff
@@ -324,6 +324,10 @@ func (s *Server) ListRemoteRepositoriesFromProvider(
 	providerName := in.GetContext().GetProvider()
 	provs, err := s.providerStore.GetByNameAndTrait(ctx, projectID, providerName, db.ProviderTypeRepoLister)
 	if err != nil {
+		var userErr *util.NiceStatus
+		if errors.As(err, &userErr) {
+			return nil, util.UserVisibleError(codes.NotFound, "no suitable provider found, please enroll a provider")
```

Comment:
If this is already a nice status, why not return that?

Reply:
Because we call:
 - ListRemoteRepositoriesFromProvider
 - GetByNameAndTrait
 - findProvider
 - filteredResultNotFoundError which returns a NiceStatus saying `provider not found with filters`. On the one hand this is a NiceStatus so it is suitable for rendering by clients, on the other hand it doesn't sound like an error that's very actionable by the user.

I wonder if we should rather not return a NiceStatus from a utility function or convert the NiceStatus from filteredResultNotFoundError to a bespoke `error` that would be then converted to a NiceStatus in the handlers.

Reply:
I think we shouldn't. We should return relevant internal errors and only do gPRC status errors on the handler functions.